Lib/test/test_functools.py

```diff
@@ -1271,6 +1271,19 @@ def f(x):
         self.assertEqual(f(20), '.20.')
         self.assertEqual(f.cache_info().currsize, 10)
 
+    def test_lru_bug_36650(self):
```

Comment:
Putting myself in the mind of my future self, I wonder if if this test should, instead of focusing on the bug, simply declare the intended behavior, something like:

```
def test_lru_empty_kw_arguments_same_as_absent():
    """
    A call with no keyword arguments should match a call with
    empty keyword arguments.
    
    Avoids an unexpected cache miss on equivalent calls.
    See #36650 for background.
    """
    ...
```

This approach would avoid making reference to incidental details like "a broken doctest", and makes reference to the issue for those interested in knowing why this might matter, but otherwise describes the primary intention. It also avoids a comment that "C version was treating", which seemingly implies the previous behavior has been around a long time (and not just since 3.7.3).

I notice you aren't using docstrings, so an equivalent comment-based version might look like this:

```
def test_lru_empty_kw_arguments_same_as_absent():
    # A call with no keyword arguments should match a call with
    # empty keyword arguments. Avoids an unexpected cache miss
    # on equivalent calls. See #36650 for background.
    ...
```

Reply:
The test naming follows our usual style in the standard library tests. It is debatable whether that is what should have been done, but it is the norm.  I will elide the reference to the doctest.